### ISSUE TITLE
pytest and pytest-xdist plugin with `--snapshot-update -n auto` deletes all snapshots on parallel executions

### ISSUE THREAD

**Issue by Sharu95** [pytest-dev/pytest#9600]
# Description
Currently working on optimizing some pipelines where I discovered that pytest + pytest-xdist behaves different with the `-n` flag combined with `--snapshot-update`. The issue arised while using `tox`, but is replicated with only `pytest`, `pytest-xdist` and `syrupy`. A minimal codebase is provided to replicate the results further down. 

**Actual outcome**
All snapshots (new and old) are deleted when tests are run with `--snapshot-update -n auto`
**Expected outcome**
All old snapshots should not be deleted and a new snapshot should be added with `--snapshot-update -n auto` 

# Steps to reproduce

1. Create initial snapshots with `pytest --snapshot-update`
2. Add a new test to the test-file 
3. Run `pytest --snapshot-update -n auto` to run tests in parallel, which deletes snapshots

**Screenshot of steps**
![steps to reproduce](https://user-images.githubusercontent.com/14542723/152400673-8a6889c4-beb4-4206-9767-40ee4827e9b7.png)

## System information
|Name|Version|
|--|--|
python|3.7.12
pip|21.3.1
syrupy|1.7.3
pytest-xdist|2.5.0
pytest|6.2.5
MacBook Pro 2016|MacOS Big Sur 11.6

[piplist.txt](https://github.com/pytest-dev/pytest/files/7997024/piplist.txt)
[requirements.txt](https://github.com/pytest-dev/pytest/files/7997025/requirements.txt)
Full code/minimum example: [codebase.zip](https://github.com/pytest-dev/pytest/files/7997052/codebase.zip)

**Comment by RonnyPfannschmidt** [comment]
duplicate of https://github.com/tophat/syrupy/issues/535
